ssl/statem/statem_clnt.c

```diff
@@ -175,7 +175,7 @@ int ossl_statem_client_read_transition(SSL *s, int mt)
             } else {
                 ske_expected = key_exchange_expected(s);
                 if (ske_expected < 0)
-                    return 0;
+                    goto err;
```

Comment:
Hm, key_exchange_expected (static method) documents it may return -1, but never does. Could you just make it a 1/0 boolean instead to simplify the logic?